src/PackageManagement/IDE/PackageRestoreManager.cs

```diff
@@ -347,19 +350,19 @@ private NuGetPackageManager GetNuGetPackageManager(string solutionDirectory)
         /// performs package restore
         /// Note that this method should only Dequeue from the concurrent queue and not Enqueue
         /// </summary>
-        private static async Task PackageRestoreRunnerAsync(ConcurrentQueue<Packaging.PackageReference> packageReferencesQueue,
+        private static async Task<bool> PackageRestoreRunnerAsync(ConcurrentQueue<Packaging.PackageReference> packageReferencesQueue,
             PackageRestoreContext packageRestoreContext,
             INuGetProjectContext nuGetProjectContext)
         {
             Packaging.PackageReference currentPackageReference = null;
+            bool restoreResult = true;
             while (packageReferencesQueue.TryDequeue(out currentPackageReference))
             {
                 var result = await RestorePackageAsync(currentPackageReference, packageRestoreContext, nuGetProjectContext);
-                if (result)
-                {
-                    packageRestoreContext.SetRestored();
```

Comment:
could you search the code to see if SetRestored() is still used? If not, we should remove it.